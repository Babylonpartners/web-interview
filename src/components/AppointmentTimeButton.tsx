import * as React from 'react'
import { activeButtonStyles } from '../styles/styles';

interface Props {
    handleChange: any
    selectedAppointmentTime: any
    slot: any
    monthNames: any
}

export class AppointmentTimeButton extends React.PureComponent<Props> {
    render() {

        return (
            <>
            {!this.props.slot || !this.props.slot.time ?
            null
            :
            <>
            {this.props.selectedAppointmentTime === this.props.slot.time
            ?
            <button
                data-testid="selectedTimeButton"
                style={activeButtonStyles}
                className="button"
                value={this.props.slot.time}
                onClick={this.props.handleChange}
            >
            {new Date(this.props.slot.time).toLocaleDateString()
            === 
            new Date().toLocaleDateString()
            ?
                <>Today {new Date(this.props.slot.time).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})} </>
            :
                <> {new Date(this.props.slot.time).getDate()} {this.props.monthNames[new Date(this.props.slot.time).getMonth()]}, {new Date(this.props.slot.time).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})} </>
            }  
            </button>
            :
            <button
                data-testid="unselectedTimeButton"
                style={activeButtonStyles}
                className="white-button"
                value={this.props.slot.time}
                onClick={this.props.handleChange}
            >
            {new Date(this.props.slot.time).toLocaleDateString()
            === 
            new Date().toLocaleDateString()
            ?
                <>Today {new Date(this.props.slot.time).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})} </>
            :
                <>{new Date(this.props.slot.time).getDate()} {this.props.monthNames[new Date(this.props.slot.time).getMonth()]}, {new Date(this.props.slot.time).toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'})} </>
            }  
            </button>
            }

            </>
            }
            </>
        );
}}