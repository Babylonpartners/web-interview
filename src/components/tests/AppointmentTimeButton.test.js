import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { AppointmentTimeButton } from '../AppointmentTimeButton'

afterEach(() => {
  cleanup()
  console.error.mockClear()
})

console.error = jest.fn()

const slot = {
  id: 1,
  consultantType: ['gp'],
  appointmentType: ['audio', 'video'],
  time: '2019-11-27T10:11:00.000Z',
}

const monthNames = [
  'Jan',
  'Feb',
  'Mar',
  'Apr',
  'May',
  'Jun',
  'Jul',
  'Aug',
  'Sep',
  'Oct',
  'Nov',
  'Dec',
]

test('<AppointmentTimeButton />', () => {
  const { container } = render(<AppointmentTimeButton />)
  expect(container.firstChild).toMatchSnapshot()
  expect(console.error).not.toHaveBeenCalled()
})

test('<AppointmentTimeButton /> with slot', () => {
  const { getByTestId } = render(
    <AppointmentTimeButton slot={slot} monthNames={monthNames} />
  )
  expect(console.error).not.toBeCalled()
  expect(slot.id).toEqual(1)
  expect(slot.consultantType).toEqual(['gp'])
  expect(slot.appointmentType).toEqual(['audio', 'video'])
  expect(slot.time).toEqual('2019-11-27T10:11:00.000Z')
  expect(getByTestId('unselectedTimeButton')).toBeTruthy()
})
