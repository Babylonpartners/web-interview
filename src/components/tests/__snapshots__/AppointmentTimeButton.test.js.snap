// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppointmentTimeButton /> 1`] = `null`;
